cpp/src/sampling/random_walks_impl.cuh

```diff
@@ -139,7 +185,7 @@ struct uniform_selector {
 
 template <typename weight_t>
 struct biased_selector {
-  uint64_t seed_{0};
+  raft::random::RngState rng_state_;
```

Comment:
This will copy-construct a new `RngState` object. You need to store a reference (or a pointer) to the passed `rng_state` object.